spark/src/test/scala/org/apache/spark/sql/delta/DeltaProtocolVersionSuite.scala

```diff
@@ -2397,6 +2397,60 @@ trait DeltaProtocolVersionSuiteBase extends QueryTest
     }
   }
 
+  test("Can construct a protocol with column mapping only in writer features") {
+    // Column mapping should appear in both reader and writer features. However,
+    // we allow the construction of invalid protocols such as the one below to allow backward
```

Comment:
Why would the one below be invalid? Is it because `readerFeatures` should have been `None`?

Reply:
But the reader version below is 2. In that case the readers features should not be set, right?

Reply:
Both (2, 5) and (2, 7, columnMapping, ...) are valid protocols. The first is in legacy form and the later in the table features form.

Once we are in the table features territory all features need to be explicitly set. In a sense, yes (2, 7) implies there is column mapping only because column mapping is the only reader feature with 2. However, the semantics of the table feature protocols is to explicitly state all supported features.

I will add an extra tests to also include the none case.